### ISSUE TITLE
Warning due to shadowing of clojure.core/update-vals

### ISSUE THREAD

**Issue by alexhall** [clj-kondo/clj-kondo#2279]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**: RELEASE

**macro usage**: N/A

**platform**: JVM

**editor**: N/A

**problem**

Running `clj-kondo` from the latest RELEASE in Maven yields the following warning message:

```
WARNING: update-vals already refers to: #'clojure.core/update-vals in namespace: clj-kondo.impl.analysis.java, being replaced by: #'clj-kondo.impl.utils/update-vals
```

**repro**: N/A

**config**: N/A

**expected behavior**: No warnings printed

Accompanying PR to follow
